src/browser-polyfill.js

```diff
@@ -13,7 +13,7 @@ if (typeof browser === "undefined") {
   // never actually need to be called, this allows the polyfill to be included
   // in Firefox nearly for free.
   const wrapAPIs = () => {
-    const apiMetadata = {/* include("api-metadata.json") */};
+    const apiMetadata = require("../api-metadata.json");
```

Comment:
> But the `require` will not work correctly if the source version is loaded accidentally instead of the build version as a plain tag script added to an extension page, or as an AMD module.

Yes, that's the point :) Instead of silently failing because the following line is valid JS, there would be an `Uncaught ReferenceError`. Well, if both `module` and `require` are defined globally, there could still be a silent failure, but this seems unlikely.

```js
    const apiMetadata = {/* include("api-metadata.json") */};
```

> We can prevent the "CommonJS" loader (and bundler) to pick the wrong file by changing the "main" property in the package.json to the dist file.

Maybe we're misunderstanding each other here, but bundlers like browserify and webpack automatically make sure that `require`/`module`/etc are defined correctly, so they should use `src/browser-polyfill.js`. Of course, if a user wanted to `npm install webextension-polyfill`, then use a script tag to include the polyfill, they could be slightly misled by seeing `"main": "src/browser-polyfill.js"` in package.json, but I think that can be easily handled in documentation.

> Let's wait a feedback from @kmaglione on this (but I vote for switching back to the {/* include("...") */} ;-)

Agreed on waiting for feedback :) Another way to go about this would be to switch back to `{/* include("...") */}`, but then change it to something that throws an error if it's not replaced. That would also make it more obvious to the user if they accidentally use the src file in a script tag.